src/lando/main/models/repo.py

```diff
@@ -99,6 +100,28 @@ def raise_if_not(self, repo_scm):
         if repo_scm != self.scm:
             raise self._method_not_supported_for_repo_error
 
+    @property
+    def hg_repo_is_initialized(self):
+        self.raise_if_not(self.HG)
+        try:
+            self.hg._open()
+        except hglib.error.ServerError:
+            logger.info(f"{self} appears to be not initialized.")
+            return False
+        else:
+            return True
+
+    def hg_repo_prepare(self):
+        self.raise_if_not(self.HG)
+        if not self.hg_repo_is_initialized:
+            Path(self.system_path).mkdir(parents=True, exist_ok=True)
```

Comment:
Seems like `self.system_path` should already be a `Path`, based on the variable name.

Reply:
This is not correct, it is a string (see the [docs](https://docs.djangoproject.com/en/5.1/ref/models/fields/#filepathfield)). There isn't a convention of a variable ending in `_path` to be of a certain type, it is referring to what the string represents (i.e., a path and not a filename, for example.).